csml_engine/src/db_connectors/dynamodb/memories.rs

```diff
@@ -184,3 +220,70 @@ pub fn get_memories(
 
     Ok(serde_json::json!(map))
 }
+
+fn get_memory_batches_to_delete(
+    client: &Client,
+    range: String,
+    db: &mut DynamoDbClient,
+) -> Result<(), EngineError> {
+    let mut pagination_key = None;
+
+    // retrieve all memories from dynamodb
+    loop {
+        let data = query_memories(client, range.clone(), "range", None, db, 25, pagination_key)?;
+
+        // The query returns an array of items (max 10, based on the limit param above).
+        // If 0 item is returned it means that there is no open conversation, so simply return None
+        // , "last_key": :
+        let items = match data.items {
+            None => return Ok(()),
+            Some(items) if items.len() == 0 => return Ok(()),
+            Some(items) => items.clone(),
+        };
+
+        let mut write_requests = vec![];
+
+        for item in items {
+            let mem: Memory = serde_dynamodb::from_hashmap(item.to_owned())?;
+
+            let key = serde_dynamodb::to_hashmap(&DynamoDbKey {
+                hash: Memory::get_hash(client),
+                range: Memory::get_range(&mem.key, &mem.id),
+            })?;
+
+            write_requests.push(WriteRequest {
+                delete_request: Some(DeleteRequest { key }),
+                put_request: None,
+            });
+        }
+
+        let request_items = [(get_table_name()?, write_requests)]
+        .iter()
+        .cloned()
+        .collect();
+
+        let input = BatchWriteItemInput {
+            request_items,
+            ..Default::default()
+        };
+
+        execute_batch_write_query(db, input)?;
+
+        pagination_key = data.last_evaluated_key;
+        if let None = &pagination_key {
+            return Ok(())
+        }
+    }
+}
+
+pub fn delete_client_memories(client: &Client, db: &mut DynamoDbClient) -> Result<(), EngineError> {
+    get_memory_batches_to_delete(client, format!("memory#"), db)
+}
+
+pub fn delete_client_memory(
+    client: &Client,
+    key: &str,
+    db: &mut DynamoDbClient,
+) -> Result<(), EngineError> {
+    get_memory_batches_to_delete(client, format!("memory#{}", key), db)
```

Comment:
This may remove other memories as well, which is dangerous.

if we have:
- memory#tutu#ID1
- memory#tutututututu#ID2

Both will be deleted if I query begins_with(memory#tutu).

We should do:
- begins_with(memory#tutu#) (with the final #) OR exactly equals memory#tutu